Move typedargs to py3 only

mattrunchey opened the pull request:
Remove past and future libraries

Clean up some pep8 stuff

Move to azure pipelines. Keep travis for pypi deployment though.

timburke commented on typedargs/annotate.py:
```diff
@@ -86,7 +82,7 @@ def get_help(func):
 
     help_text += "\nArguments:\n"
-    for key, info in iteritems(func.metadata.annotated_params):
+    for key, info in iter(func.metadata.annotated_params.items()):
```
I think `iter` here is unnecessary since `items()` returns an iterable already.

I didn't realize we used `iteritems` and `viewitems` separately but I never made a distinction between the two and we use them interchangeably.

timburke replied:
Both `viewitems()` and `iteritems()` map to `.items()` on python 3:
https://python-future.org/_modules/future/utils.html

mattrunchey replied:
ah. I was using some random PEP doc and just dropped in the equivalents. https://legacy.python.org/dev/peps/pep-0469/ 

Good to know i can just drop the iter part.

mattrunchey commented:
Think it's good to go. Removed the travis build as a required check and added the azure check instead. Deployment should still happen on a master commit w/ tag.

Also added the fun badge and link to docs in the README.
